src/provider/APIHandler.ts

```diff
@@ -81,6 +77,24 @@ export class APIHandler<T extends Enum> {
         return this._providerChannel.connections;
     }
 
+    public dispatch(to: Identity, action: string, payload: any): Promise<any> {
+        return this._providerChannel.dispatch(to, action, payload).catch(error => {
+            // Log and re-throw
+            console.error(`Error when dispatching '${action}' to ${to.uuid}/${to.name}`, payload);
+            throw error;
+        });
+    }
+
+    public publish(action: string, payload: any): Promise<any>[] {
+        const connections = this._providerChannel.connections.slice();
+        return this._providerChannel.publish(action, payload).map((promise, index) => promise.catch(error => {
+            // Log and re-throw
+            // We don't know which connection had the error, but assume that the indices of the promises match the indices of the channel connections.
+            console.error(`Error when publishing '${action}' (probably from connection ${index + 1}/${connections.length}: ${connections[index].uuid} / ${connections[index].name})\n`, payload);
+            throw error;
```

Comment:
alternatively (maybe a bit uglier but should save you from that slice() and the assumption:
`const catchHandler = ((uuid, name, error) => { <catch goes here body> }).bind(undefined, connectionUuid, connectionUuid)`
then
`promise.catch(catchHandler);`

Reply:
well, then I realised that would mean implementing your own publish :) ignore that.